shared/src/question-generators/time/pseudoCodeUtils.ts

```diff
@@ -0,0 +1,94 @@
+/*
+This package works, so we could use this for coloring the pseudocode
+https://www.w3schools.com/tags/ref_colornames.asp
+*/
+
+export type PseudoCodeString = Array<
+  string | PseudoCodeVariable | PseudoCodeFunctionName | PseudoCodePrintString
+>
+
+export type PseudoCodeVariable = { variable: string }
+export type PseudoCodeFunctionName = { functionName: string }
+export type PseudoCodePrintString = { printString: string }
+
+export type PseudoCodePrint = { print: PseudoCodeString }
+export type PseudoCodeCall = { functionName: string; args: Array<PseudoCodeString> }
+// return as it is a keyword in TS
+export type PseudoCodeReturn = { returnValue: PseudoCodeString }
+
+export type PseudoCodeBreak = { breakState: true }
+export type PseudoCodeContinue = { continueState: true }
+// export type PseudoCodeTrue = { true: true }
+// export type PseudoCodeFalse = { false: true }
+
+export type PseudoCodeAssignment = { variable: string; value: PseudoCodeString }
+
+export type PseudoCodeIf = {
+  if: {
+    condition: PseudoCodeString
+    then: PseudoCodeState | PseudoCodeBlock | null
```

Comment:
`PseudoCodeBlock` already contains `PseudoCodeState` and the `then` part should probably never be empty.

Reply:
I added the `null`, so it's more simple to create an if. 

So you can create it straight forward and not reversed. But I can take it out if you want? It's not that important

Reply:
mh, I see the point. It's probably not that important. But it looks like it is allowed to be empty but after the generator finished some of the properties should not be null. We can also ensure this by reviewing. Maybe @holgerdell has an opinion on this.